frontend/src/metabase/query_builder/components/notebook/steps/ExpressionStep.tsx

```diff
@@ -0,0 +1,53 @@
+import React from "react";
+
+import ExpressionWidget from "metabase/query_builder/components/expressions/ExpressionWidget";
+import { NotebookStepUiComponentProps } from "metabase/query_builder/components/notebook/lib/steps.types";
+import { ExpressionValue } from "metabase/query_builder/components/expressions/expressions.types";
+import ClauseStep from "./ClauseStep";
+
+type ExpressionStepProps = NotebookStepUiComponentProps;
+
+const ExpressionStep = ({
+  color,
+  query,
+  updateQuery,
+  isLastOpened,
+  reportTimezone,
+}: ExpressionStepProps): JSX.Element => {
+  const items = Object.entries(query.expressions()).map(
+    ([name, expression]) => ({ name, expression }),
+  );
+
+  return (
+    <ClauseStep<{
```

Comment:
nit: It seems that it can be removed because TS infers the type from `items`. Tried it locally - seems to be working fine